iOS/PacketTunnelProvider/NetworkProtection/NetworkProtectionPacketTunnelProvider.swift

```diff
@@ -444,61 +445,40 @@ final class NetworkProtectionPacketTunnelProvider: PacketTunnelProvider {
-                                                       authEndpointService: authService)
-            self.accountManager = accountManager
-            let featureVisibility = NetworkProtectionVisibilityForTunnelProvider(accountManager: accountManager)
-            let accessTokenProvider: () -> String? = {
-                if featureVisibility.shouldMonitorEntitlement() {
-                    return { accountManager.accessToken }
-                }
-                return { nil }
-            }()
-            tokenHandler = NetworkProtectionKeychainTokenStore(accessTokenProvider: accessTokenProvider)
-            entitlementsCheck = { return await Self.entitlementCheck(accountManager: accountManager) }
-            self.subscriptionManager = nil
-        } else {
-            // MARK: Subscription V2
+        // Configure V2 for migration
+
+        let authEnvironment: OAuthEnvironment = subscriptionEnvironment.serviceEnvironment == .production ? .production : .staging
+        let authService = DefaultOAuthService(baseURL: authEnvironment.url,
+                                              apiService: APIServiceFactory.makeAPIServiceForAuthV2(withUserAgent: DefaultUserAgentManager.duckDuckGoUserAgent))
+
+        // keychain storage
+        let subscriptionAppGroup = Bundle.main.appGroup(bundle: .subs)
+        let tokenStorage = SubscriptionTokenKeychainStorageV2(keychainType: .dataProtection(.named(subscriptionAppGroup))) { accessType, error in
+            let parameters = [PixelParameters.privacyProKeychainAccessType: accessType.rawValue,
+                              PixelParameters.privacyProKeychainError: error.localizedDescription,
+                              PixelParameters.source: KeychainErrorSource.vpn.rawValue,
+                              PixelParameters.authVersion: KeychainErrorAuthVersion.v2.rawValue]
+            DailyPixel.fireDailyAndCount(pixel: .privacyProKeychainAccessError,
+                                         pixelNameSuffixes: DailyPixel.Constant.legacyDailyPixelSuffixes,
+                                         withAdditionalParameters: parameters)
+        }
+        let legacyAccountStorage = SubscriptionTokenKeychainStorage(keychainType: .dataProtection(.named(subscriptionAppGroup)))
+        let authClient = DefaultOAuthClient(tokensStorage: tokenStorage,
+                                            legacyTokenStorage: legacyAccountStorage,
+                                            authService: authService)
+        let pixelHandler = AuthV2PixelHandler(source: .systemExtension)
+        subscriptionAuthMigrator = AuthMigrator(oAuthClient: authClient,
+                                                    pixelHandler: pixelHandler,
+                                                    isAuthV2Enabled: settings.isAuthV2Enabled)
+        Self.shouldUseAuthV2 = subscriptionAuthMigrator.isReadyToUseAuthV2
```

Comment:
IMHO  the issue is here - the extension shouldn't migrate to authV2 on its own, but rely on the parent app to decide all this.